src/Pages/CloudAccountsPage/__tests__/CloudAccountsPage.test.tsx

```diff
@@ -0,0 +1,157 @@
+import { renderWithRouter } from '../../../utils/testing/customRender';
+import { screen, waitFor } from '@testing-library/react';
+import React from 'react';
+import { CloudAccountsPage } from '../CloudAccountsPage';
+import { ManipulatableQueryWrapper } from '../../../Components/util/testing/ManipulatableQueryWrapper';
+import { useCloudAccounts } from '../../../hooks/api/useCloudAccounts';
+import { useRbacPermission } from '../../../hooks/util/useRbacPermissions';
+
+const mockNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+  __esModule: true,
+  ...jest.requireActual('react-router-dom'),
+  Navigate: () => {
+    mockNavigate();
+    return <div data-testid="navigate" />;
+  },
+}));
+
+jest.mock('../../../hooks/api/useCloudAccounts', () => ({
+  useCloudAccounts: jest.fn(),
+}));
+jest.mock('../../../hooks/util/useRbacPermissions', () => ({
+  useRbacPermission: jest.fn(),
+}));
+jest.mock('../../../Components/CloudAccounts/CloudAccountsTable', () => ({
+  CloudAccountsTable: () => <div>Cloud Accounts Table</div>,
+}));
+jest.mock('../../../Components/CloudAccounts/CloudAccountsToolbar', () => ({
+  CloudAccountsToolbar: () => <div>Cloud Accounts Toolbar</div>,
+}));
+jest.mock('../../../Components/CloudAccounts/CloudAccountsPagination', () => ({
+  CloudAccountsPagination: () => <div>Pagination</div>,
+}));
+jest.mock('../../../Components/CloudAccounts/NoCloudAccounts', () => ({
+  NoCloudAccounts: () => <div>No cloud accounts</div>,
+}));
+jest.mock('../../../Components/util/Loading', () => ({
+  Loading: () => <div>Loading</div>,
+}));
+jest.mock('@redhat-cloud-services/frontend-components/Unavailable', () => ({
+  Unavailable: () => <div>Unavailable</div>,
+}));
```

Comment:
Issue: Instead of doing all of this mocking, manipulate the query data via `ManipulateableQueryWrapper` so that you can render the actual components. See the `GoldImagesPage.test.tsx` for examples